test/common/test_eigen.cpp

```diff
@@ -680,6 +680,21 @@ TEST (PCL, eigen33d)
   const Scalar epsilon = 2e-5;
   const unsigned iterations = 1000000;
 
+  // special case
+  r_matrix = Eigen::Matrix<Scalar, 3, 1>(3, 2, 1).asDiagonal();
+  c_matrix = r_matrix;
+
+  eigen33(r_matrix, r_vectors, r_eigenvalues);
+  // check if the main vector is the positive Z direction. In this case it is equal to (0,0,1)
+  EXPECT_LE(fabs(r_vectors.col(0)[0]), epsilon);
```

Comment:
Semantically, what you are interested is in `EXPECT_NEAR`.
https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#floating-point-macros

Also, prefer the use of `std::abs` over `fabs`.

Reply:
This was copy pasted from the other PR, so I didn't glance at the code. At least, this tells that the reviewing policy has gotten better :)

I'll perform the necessary edits.